Tealium/src/ios/tealium_int.m

```diff
@@ -14,54 +14,34 @@ - (void) init: (CDVInvokedUrlCommand*)command {
+    NSString *timestamp = [NSString stringWithFormat:@"%li", tiLong];
+    NSString *osVersion = [[UIDevice currentDevice] systemVersion];
+
 	@try {
 	    arguments = [command.arguments objectAtIndex:0];
 	    accountName = [arguments objectForKey:@"account"];
 	    profileName = [arguments objectForKey:@"profile"];
 	    environmentName = [arguments objectForKey:@"environment"];
-		
-		id disableHTTPSVal = [arguments objectForKey:@"disableHTTPS"];
-		id suppressLogsVal = [arguments objectForKey:@"suppressLogs"];
-        id disableExceptionHandlingVal = [arguments objectForKey:@"disableExceptionHandling"];
-		
-		BOOL disableHTTPS = disableHTTPSVal == nil ? NO : [disableHTTPSVal boolValue];
-		BOOL suppressLogs = suppressLogsVal == nil ? YES : [suppressLogsVal boolValue];
-        BOOL disableExceptionHandling = disableExceptionHandlingVal == nil ? NO : [disableExceptionHandlingVal boolValue];
-				
-		options = 
-			(disableHTTPS ? TLDisableHTTPS : 0) |
-			(suppressLogs ? TLSuppressLogs : TLDisplayVerboseLogs) |
-			(disableExceptionHandling ? disableExceptionHandling : 0);
-        
-        // Below functions did not work as regular methods during Cordova build - added inline below
-        
-        // timestamp
-        NSTimeInterval ti = [[NSDate date] timeIntervalSince1970];
-        long tiLong = (long)ti;
-        NSString *timestamp = [NSString stringWithFormat:@"%li", tiLong];
-        
-        // mobile url override
-        NSString *osVersion         = [[UIDevice currentDevice] systemVersion];
-        NSString *mobileURLString = [NSString stringWithFormat:Tealium_MobileBaseURL,accountName, profileName, environmentName, TealiumDSK_Platform, Tealium_Platform,  TealiumDSK_OSVersion, osVersion, TealiumDSK_LibraryVersion, Tealium_LibVersion, TealiumDSK_TimestampUnix, timestamp];
-        
-        customData = @{TealiumDSK_OverrideUrl : mobileURLString};
-			
+        instanceName = [arguments objectForKey:@"instance"];
 	}
 	@catch (NSException *exception) {
         pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
         [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
         return;
 	}
-    
-    if (accountName != nil && [accountName length] > 0) {
-        [Tealium initSharedInstance: accountName
-                            profile: profileName
-                             target: environmentName
-							options: options
-                   globalCustomData: customData];
+    // generate mobile.html override string
+    NSString *mobileURLString = [NSString stringWithFormat:Tealium_MobileBaseURL, accountName, profileName, environmentName, TEALDataSourceKey_Platform, Tealium_Platform, TEALDataSourceKey_DeviceOSVersion, osVersion, TEALDataSourceKey_LibraryVersion, Tealium_LibVersion, TEALDataSourceKey_TimestampUnix, timestamp];
+
+    if (accountName != nil && [accountName length] > 0 && instanceName != nil && [instanceName length] > 0) {
```

Comment:
Added very granular checks for account/profile/instance name to provide useful feedback on invalid arguments.